openrouteservice/src/main/java/org/heigit/ors/matrix/algorithms/core/CoreMatrixAlgorithm.java

```diff
@@ -0,0 +1,547 @@
+        if(hasDistances)
+            newDistances = new float[distances.length];
+        if(hasWeights)
+            newWeights = new float[weights.length];
+
+        int i = 0;
+        int srcSize = srcData.size();
+        int dstSize = dstData.size();
+        for (int dst = 0; dst < dstSize; dst++){
+            for(int src = 0; src < srcSize; src++){
+                if(hasTimes)
+                    newTimes[dst + src * dstSize] = times[i];
+                if(hasDistances)
+                    newDistances[dst + src * dstSize] = distances[i];
+                if(hasWeights)
+                    newWeights[dst + src * dstSize] = weights[i];
+                i++;
+            }
+        }
+        return new float[][]{newTimes, newDistances, newWeights};
+    }
+
+    private void extractMetrics(MatrixLocations srcData, MatrixLocations dstData, float[] times, float[] distances, float[] weights) throws Exception {
+        AveragedMultiTreeSPEntry[] destTrees = new AveragedMultiTreeSPEntry[dstData.size()];
+        for (int i = 0; i < dstData.size(); i++)
+            destTrees[i] = targetMap.get(dstData.getNodeIds()[i]);
+
+        AveragedMultiTreeSPEntry[] originalDestTrees = new AveragedMultiTreeSPEntry[dstData.size()];
+
+        int j = 0;
+        for (int i = 0; i < dstData.size(); i++) {
+            if (dstData.getNodeIds()[i] != -1) {
+                originalDestTrees[i] = destTrees[j];
+                ++j;
+            } else {
+                originalDestTrees[i] = null;
+            }
+        }
+        pathMetricsExtractor.setSwap(swap);
+        pathMetricsExtractor.calcValues(originalDestTrees, srcData, dstData, times, distances, weights);
+    }
+
+    boolean isCoreNode(int node) {
+        return chGraph.getLevel(node) >= coreNodeLevel;
+    }
+
+    boolean considerTurnRestrictions() {
+        if (!hasTurnWeighting)
+            return false;
+        return true;
+    }
+
+    /**
+     * Check whether the turnWeighting should be in the inORS mode. If one of the edges is a virtual one, we need the original edge to get the turn restriction.
+     * If the two edges are actually virtual edges on the same original edge, we want to disable inORS mode so that they are not regarded as u turn,
+     * because the same edge id left and right of a virtual node results in a u turn
+     * @param iter
+     * @param currEdge
+     * @return
+     */
+    private boolean isInORS(EdgeIteratorState iter, AveragedMultiTreeSPEntry currEdge) {
+        return currEdge.getEdge() == iter.getEdge() || EdgeIteratorStateHelper.getOriginalEdge(iter) != EdgeIterator.NO_EDGE;
+    }
+
+    private boolean isInORS(EdgeIteratorState iter, MultiTreeSPEntryItem currEdgeItem) {
+        return currEdgeItem.getEdge() == iter.getEdge() || currEdgeItem.getOriginalEdge() != EdgeIteratorStateHelper.getOriginalEdge(iter);
+    }
+
+    public void setEdgeFilter(CoreDijkstraFilter additionalEdgeFilter) {
```

Comment:
Any use case for this public method? For now it seems to be only called from the private `initFilter` method. If currently it's not used otherwise then probably it could go away.

Reply:
If it is used in `initFilter`, why should I remove it? I find that using methods to do assignments is preferred over pure `x=y`, especially in cases like this where there are multiple edgeFilters in play. It is good to separate functionality.

Reply:
i removed everything